src/garage/torch/algos/vpg.py

```diff
@@ -183,39 +225,60 @@ def _compute_loss(self, itr, obs, actions, rewards, valids, baselines):
+        rewards_flat = torch.cat(filter_valids(rewards, valids))
+        advantages_flat = self._compute_advantage(rewards, valids, baselines)
 
-        policy_entropies = self._compute_policy_entropy(obs)
+        return self._compute_loss(obs_flat, actions_flat, rewards_flat,
+                                  advantages_flat)
 
-        if self._maximum_entropy:
-            rewards += self._policy_ent_coeff * policy_entropies
+    def _compute_loss(self, obs, actions, rewards, advantages):
+        """Compute mean value of loss.
+
+        Args:
+            obs (torch.Tensor): Observation from the environment.
+            actions (torch.Tensor): Predicted action.
+            rewards (torch.Tensor): Feedback from the environment.
+            advantages (torch.Tensor): Expected rewards over the actions.
+
+        Returns:
+            torch.Tensor: Calculated mean value of loss
+
+        """
+        objective = self._compute_objective(advantages, obs, actions, rewards)
 
+        return -objective.mean()
+
+    def _compute_advantage(self, rewards, valids, baselines):
+        """Compute mean value of loss.
+
+        Args:
+            rewards (torch.Tensor): Feedback from the environment.
+            valids (list[int]): Array of length of the valid values.
+            baselines (torch.Tensor): Value function estimation at each step.
+
+        Returns:
+            torch.Tensor: Calculated advantage values given rewards and
+                baselines.
+
+        """
         advantages = compute_advantages(self.discount, self._gae_lambda,
                                         self.max_path_length, baselines,
                                         rewards)
+        advantage_flat = torch.cat(filter_valids(advantages, valids))
 
         if self._center_adv:
-            means, variances = list(
-                zip(*[(valid_adv.mean(), valid_adv.var())
-                      for valid_adv in filter_valids(advantages, valids)]))
-            advantages = F.batch_norm(advantages.t(),
-                                      torch.Tensor(means),
-                                      torch.Tensor(variances),
-                                      eps=self._eps).t()
+            means = advantage_flat.mean()
+            variance = advantage_flat.var()
+            advantage_flat = (advantage_flat - means) / (variance + 1e-8)
 
         if self._positive_adv:
-            advantages -= advantages.min()
+            advantage_flat -= advantage_flat.min()
 
-        objective = self._compute_objective(advantages, valids, obs, actions,
-                                            rewards)
+        return advantage_flat
 
-        if self._entropy_regularzied:
-            objective += self._policy_ent_coeff * policy_entropies
-
-        valid_objectives = filter_valids(objective, valids)
-        return -torch.cat(valid_objectives).mean()
-
-    def _compute_kl_constraint(self, obs):
+    def compute_kl_constraint(self, obs):
```

Comment:
is there a reason this needs to be public?

Reply:
Same as compute_policy_entropy, this function is used in MAML class

Reply:
see my other comment.